OrbitGl/TimeGraph.cpp

```diff
@@ -914,20 +914,36 @@ void TimeGraph::JumpToNeighborBox(TextBox* from, JumpDirection jump_direction,
   }
   auto function_address = from->GetTimer().m_FunctionAddress;
   auto current_time = from->GetTimer().m_End;
+  auto thread_id = from->GetTimer().m_TID;
   if (jump_direction == JumpDirection::kPrevious) {
-    if (jump_scope == JumpScope::kSameThread) {
-      goal = FindPrevious(from);
-    }
-    if (jump_scope == JumpScope::kSameFunction) {
-      goal = FindPreviousFunctionCall(function_address, current_time);
+    switch (jump_scope) {
+      case JumpScope::kSameDepth:
+        goal = FindPrevious(from);
+        break;
+      case JumpScope::kSameFunction:
+        goal = FindPreviousFunctionCall(function_address, current_time);
+        break;
+      case JumpScope::kSameThreadSameFunction:
+        goal =
+            FindPreviousFunctionCall(function_address, current_time, thread_id);
+        break;
+      default:
+        break;
```

Comment:
Like discussed oob, this should be a CHECK(false). Apart from that it's fine from my side.